### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by auzias** [SSilence/selfoss#602]
Hi,

I got this error message, did any got it and pass thrgough already?

```
Internal Server Error

Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 101508879 bytes)
```

**Comment by mmehnert** [comment]
I currently experience the same issue with latest HEAD. the update process consumes memory until there is nothing left on the machine.

**Comment by mmehnert** [comment]
Last item in the debug log before running out of memory is always

    09-18-16 13:04:42 Debug destroy spout object
    09-18-16 13:04:42 Debug ---
    09-18-16 13:04:42 Debug start fetching source "reddit username (id: some_id)
    09-18-16 13:04:42 Debug spout successfully loaded: spouts\reddit\reddit2
    09-18-16 13:04:42 Debug fetch content
    09-18-16 13:04:43 Debug minimum date: 2016-08-19 13:04:43
    09-18-16 13:04:43 Debug start item fetching

**Comment by jtojnar** [comment]
@mmehnert I just managed to reproduce this. The reddit spout will require a complete re-write so here is just a [quick fix](https://github.com/jtojnar/selfoss/tree/reddit-fix). There are actually multiple bugs causing this, I only fixed two of them.

**Comment by jtojnar** [comment]
The patch was merged with #805. If the issue still occurs, please re-open.
